### ISSUE TITLE
Subpages no longer possible under mainpage

### ISSUE THREAD

**Issue by MarkShackman** [doxygen/doxygen#7274]
Regression from 1.8.14 to 1.8.16:

A \subpage under the \mainpage no longer appears in the ToC as a sub-page.

Code:
```
/**
\mainpage Main Page

   \subpage SubPage1 Sub-Page 1.
   \li \subpage SubPage2 Sub-Page 2.

\page SubPage1 Sub-Page 1 Title.

\page SubPage2 Sub-Page 2 Title.
*/
```
Output:

![visual](https://user-images.githubusercontent.com/23169476/65332713-1e939f80-dbb7-11e9-816e-e7f7b498ea63.png)

**Comment by albert-github** [comment]
Some observations:
Looking at the problem, looks like the problem occurs with a default Doxyfile with just set the `GENERATE_TREEVIEW=YES` as well (looks like the report is about the CHM output but plain HTML shows the problem as well).
Looks like version 1.8.15 is OK.
When using `\subpage` as part of a `\page` command the indentation in the tree is present.

@MarkShackman in future please also add the used doxygen configuration (as attachment or as `doxygen -x` output) to the issue.

**Comment by MarkShackman** [comment]
Doxyfile was just to generate CHM file, hence not included:

```
GENERATE_HTMLHELP      = YES
CHM_FILE               = test.chm
HHC_LOCATION           = ..\bin\hhc.exe
```
\subpage still works as expected elsewhere; issue is just within \mainpage. 
Confirm that 1.8.15 behaves as expected.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7282

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
